pages/api/auth/[...nextauth].ts

```diff
@@ -1,11 +1,24 @@
-import NextAuth from 'next-auth'
+import NextAuth, { NextAuthOptions } from 'next-auth'
 import CredentialProvider from 'next-auth/providers/credentials'
 import { dbConnect } from '../../../server/_dbConnect'
 import bcrypt from 'bcryptjs'
 import type { WithId, Document } from 'mongodb'
 
 /* eslint-disable */
-export default NextAuth({
+
+const signinUser = async (user: WithId<Document>, pwd: string) => {
+  if (!user.hashPass) {
+    throw new Error('Please enter password')
+  }
+  const isMatch = await bcrypt.compare(pwd, user.hashPass)
+
+  if (!isMatch) {
+    throw new Error('Password Incorrect')
```

Comment:
this might be completely unnecessary for edlaw but in general for safety, consider making the error less specific. For example, "Incorrect login" or "could not find an account with that information" so that people don't know it's the password specifically that's incorrect?

Reply:
so true